packages/@aws-cdk/aws-cognito-identitypool-alpha/lib/identitypool.ts

```diff
@@ -388,9 +396,14 @@ export class IdentityPool extends Resource implements IIdentityPool {
   private cognitoIdentityProviders: CfnIdentityPool.CognitoIdentityProviderProperty[] = [];
 
   /**
-   * Running count of added role attachments
+   * The list of role mappings for the Identity Pool
+   */
+  public roleMappings: IdentityPoolRoleMapping[];
+
+  /**
+   * Role attachment L2 construct for the Identity Pool
    */
-  private roleAttachmentCount: number = 0;
+  private roleAttachment: IdentityPoolRoleAttachment | null = null;
```

Comment:
I don't think this union type will be supported by JSII. Just make this optional and set it in the constructor.